src/socket.cpp

```diff
@@ -41,7 +41,7 @@ void Socket::asyncSend(const Packet& response, const std::function<void(const er
     const std::vector<uint8_t> vResponse = response.makeSendBuffer(m_secret);
     std::copy(vResponse.begin(), vResponse.end(), m_recvBuffer.begin());
```

Comment:
You use `m_recvBuffer` for both sending and receiving data. Let's call it just `m_buffer`.